apps/mobile/modules/mobile-vscode-secure-store/android/src/main/java/expo/modules/mobilevscodesecurestore/MobileVSCodeSecureStoreModule.kt

```
@@ -0,0 +1,224 @@
+
+      val tagLength = item.getInt(TAG_LENGTH_PROPERTY)
+      if (tagLength < MIN_GCM_TAG_LENGTH_BITS) {
+        throw MobileVSCodeSecureStoreException("Secure storage authentication tag is too short.")
+      }
+
+      val ciphertext = Base64.decode(item.getString(CIPHERTEXT_PROPERTY), Base64.DEFAULT)
+      val iv = Base64.decode(item.getString(IV_PROPERTY), Base64.DEFAULT)
+      val cipher = Cipher.getInstance(AES_CIPHER)
+      cipher.init(Cipher.DECRYPT_MODE, secretKey, GCMParameterSpec(tagLength, iv))
+      return@synchronized String(cipher.doFinal(ciphertext), StandardCharsets.UTF_8)
+    } catch (error: KeyPermanentlyInvalidatedException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)